core/src/main/java/org/elasticsearch/common/io/stream/StreamInput.java

```diff
@@ -419,6 +420,21 @@ public final Boolean readOptionalBoolean() throws IOException {
         return (Map<String, Object>) readGenericValue();
     }
 
+    /**
+     * Read a map of strings to string lists.
+     */
+    public Map<String, List<String>> readMapOfLists() throws IOException {
```

Comment:
The name makes the method sound generic and I think the method may have more uses if it is generic, like:

```java
    public <T, R> Map<T, List<R>> readMapOfLists(StreamInputReader<T> keyReader, StreamInputReader<R> listReader) throws IOException {
        int size = readVInt();
        if (size == 0) {
            return Collections.emptyMap();
        }
        Map<T, List<R>> map = new HashMap<>(size);
        for (int i = 0; i < size; ++i) {
            map.put(keyReader.read(this), readList(listReader));
        }
        return map;
    }
```

The method above is not tied to string keys and string lists. The caller here would use:

```java
in.readMapOfLists(StreamInput::readString, StreamInput::readString);
```